xbmc/interfaces/AnnouncementManager.cpp

```diff
@@ -25,6 +25,8 @@
 
 using namespace ANNOUNCEMENT;
 
+const std::string CAnnouncementManager::ANNOUNCEMENT_SENDER = "xbmc";
```

Comment:
So removing the xbmc from all the announcements is great, should this be taken a step further, and rather than hardcode this as "xbmc" should we use 

https://github.com/xbmc/xbmc/blob/94184731d2bb21f98d5964a268e18cc2a04ffb9d/xbmc/CompileInfo.cpp.in#L43

Reply:
Without looking at the code, I can say usually not, because the cast from char* to std::string requires an alloc anywhere the string is used, whereas instantiating as std::string upfront gives you data that can be used later without allocations.

Reply:
couldnt you just static const std::string ... = CCompileInfo::GetAppName()?

Reply:
is it specified as static: https://github.com/xbmc/xbmc/blob/1d40c5a3e702e0bcb919d1a9ee6dda360eb3661d/xbmc/interfaces/AnnouncementManager.h#L61

for linking, it has to be defined in a compilation unit with the rest of the class

Reply:
> couldnt you just static const std::string ... = CCompileInfo::GetAppName()?
> 
> The rest of this PR essentially drops the sender being passed, and this is then just assumed in the announcementmanager to be this const

See comment below.
There seem to be a bunch of addons and/or json clients that check on their side if the sender is "xbmc".
Changing from "xbmc" to GetAppName() would break all of them, guess it's not worth the trouble.